### PR TITLE
Issue 1 Implementation of MVP1

### PR DESCRIPTION
**Pull request by adamgft**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+  type = string
+}
+
+variable "mig_hostname" {
+  default = "forward-proxy-mig"
+  description = "hostname of mig"
+  type = string
+}
+
+variable "service_account" {
```
Comment (nhsy) [inline variables.tf:44]: Rename to service_account_email

File: variables.tf [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+  description = "network self link"
+  type = string
+}
+
+variable "region" {
+  description = "region"
+  type = string
+}
+
+variable "folder_id" {
```
Comment (nhsy) [inline variables.tf:21]: Why is folder_id required?

File: mig.tf [hunk 3]
```diff
@@ -1,16 +1,31 @@
...
+  tags               = ["iap"]
+
+  preemptible    = var.preemptive
+  startup_script = file("${path.module}/files/metadata-startup.sh")
+  region         = var.region
+
+  depends_on = [module.service-account]
+}
+
+module "mig" {
```
Comment (nhsy) [inline mig.tf:19]: Where are TCP port 3128 health checks defined?

File: iam.tf [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module "service-account" {
+  source  = "terraform-google-modules/service-accounts/google"
+  version = "3.0.1"
```
Comment (nhsy) [inline iam.tf:3]: Should this be version = "~> 3.0.0"

File: variables.tf [hunk 5]
```diff
@@ -1,5 +1,54 @@
...
+
+variable "mig_hostname" {
+  default = "forward-proxy-mig"
+  description = "hostname of mig"
+  type = string
+}
+
+variable "service_account" {
+  default = ""
+  description = "hostname of sa attached to forward proxy instance"
```
Comment (nhsy) [inline variables.tf:46]: Typo in description

File: iam.tf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  count   = var.service_account != "" ? 0 : 1
+
+  project_id    = var.project_id
+  names         = [local.sa_name]
+  project_roles = []
+}
+
+module "folder-iam" {
```
Comment (nhsy) [inline iam.tf:11]: Don't require this folder iam permission.

Minimum permissions required as follows:

gcloud iam service-accounts create sa-name \
  --display-name=sa-name

gcloud projects add-iam-policy-binding project-id \
  --member "serviceAccount:sa-name@project-id.iam.gserviceaccount.com" \
  --role roles/logging.logWriter

gcloud projects add-iam-policy-binding project-id \
  --member "serviceAccount:sa-name@project-id.iam.gserviceaccount.com" \
  --role roles/monitoring.metricWriter

gcloud projects add-iam-policy-binding project-id \
  --member "serviceAccount:sa-name@project-id.iam.gserviceaccount.com" \
  --role roles/monitoring.viewer

gcloud projects add-iam-policy-binding project-id \
  --member "serviceAccount:sa-name@project-id.iam.gserviceaccount.com" \
  --role roles/stackdriver.resourceMetadata.writer

**Review by nhsy — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (adamgft) [reply variables.tf:44]: changed to service_account_name as this var is the name of the sa

Comment (adamgft) [reply variables.tf:44]: change to service_account_name as this var is actually the name of the sa

File: variables.tf (hunk 2 above)
Comment (adamgft) [reply variables.tf:21]: removed

File: iam.tf (hunk 6 above)
Comment (adamgft) [reply iam.tf]: refactored to project iam and added above permissions

File: iam.tf (hunk 4 above)
Comment (adamgft) [reply iam.tf]: moved to 3.0.0

**Pull request merged**